### PR TITLE
US-3.3.2: Implement spaces number, spaces requests list and number with accept and ignore actions

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

**Comment by azayati** [comment]
[Related Social PR](https://github.com/exodev/social/pull/610)

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 1]
```diff
@@ -45,3 +45,27 @@ export function getUserInformations() {
       }
     })
 }
+
+export function getSpaces() {
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 2]
```diff
@@ -126,8 +130,23 @@
                   this.firstName = data.firstname;
                 })
       },
+      getSpacesSise() {
+        getSpaces().then(
+                (data) => {
+                  this.spaces = data.size;
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 3]
```diff
@@ -126,8 +130,23 @@
                   this.firstName = data.firstname;
                 })
       },
+      getSpacesSise() {
+        getSpaces().then(
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 4]
```diff
@@ -36,12 +36,12 @@
...
               flat>
-              <a class="white--text" @click="getSpecificCard('SpacesRequests')">
+              <a class="white--text" @click="spaceInvitation > 0 ? getSpecificCard('SpacesRequests'):toProfileStats()">
                 <v-badge class="badge-color">
                   <template v-slot:badge>
-                    <span>4</span>
+                    <span v-if="spaceInvitation > 0">{{ spaceInvitation }}</span>
                   </template>
-                  <span class="headline blue-grey--text font-weight-bold pa-1">53</span>
+                  <span class="headline blue-grey--text font-weight-bold pa-1">{{ spaces }}</span>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js [hunk 5]
```diff
@@ -45,3 +45,27 @@ export function getUserInformations() {
...
+    return fetch( '/portal/rest/v1/social/spaces?returnSize=true', {
+      method: 'GET',
+    }).then((resp) => {
+      if(resp && resp.ok) {
+        return resp.json();
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/profilStatsAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 6]
```diff
@@ -36,12 +36,12 @@
             align-center>
             <v-card
               flat>
-              <a class="white--text" @click="getSpecificCard('SpacesRequests')">
+              <a class="white--text" @click="spaceInvitation > 0 ? getSpecificCard('SpacesRequests'):toProfileStats()">
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Updated

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 7]
```diff
@@ -36,12 +36,12 @@
...
                 <v-badge class="badge-color">
                   <template v-slot:badge>
-                    <span>4</span>
+                    <span v-if="spaceInvitation > 0">{{ spaceInvitation }}</span>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue [hunk 8]
```diff
@@ -126,8 +130,23 @@
...
+      getSpacesSise() {
+        getSpaces().then(
+                (data) => {
+                  this.spaces = data.size;
+                })
+      },
+      SpaceInvitaions() {
+        getSpacesInvitations().then(
+                (data) => {
+                  this.spaceInvitation = data.size;
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/profileStats/components/UserDashbord.vue]: Done

**Pull request merged**